packages/angular_devkit/build_optimizer/src/helpers/ast-utils.ts

```diff
@@ -5,9 +5,12 @@
  * Use of this source code is governed by an MIT-style license that can be
  * found in the LICENSE file at https://angular.io/license
  */
+// tslint:disable-next-line: no-implicit-dependencies
+import * as tslib from 'tslib';
```

Comment:
Technically `tslib` should be marked as a peerDep as it is require.

Any reason why it's not at the moment?

Reply:
It definitely should be now that it is used explicitly.

Reply:
are you for peer or direct dep?

Reply:
I think peer dep is more correct but we would need to propagate it up through `build-angular` since it is a project level dependency.

Reply:
Wouldn't we get something like `@angular-devkit/build-angular > @angular-devkit/build-optimizer" has incorrect peer dependency "tslib".`?

Reply:
It would need to be a peer dep on `build-angular` as well.  That way `build-angular` satisfies `build-optimizer`.

But let's go with a direct dep for now. Since it won't cause peer dep warning changes which is problematic in a patch/minor release.

Reply:
Done with the direct dependecy